config/routes.rb

```diff
@@ -1,19 +1,35 @@
 Rails.application.routes.draw do
-
   devise_for :users
-  
-  resources :recipes
-  resources :recipe_foods
   resources :foods
-  resources :inventory_foods
-  resources :inventories
-  resources :users
+  resources :public_recipes, only: [:index]
+  resources :recipes do
+    resources :recipe_foods
+  end
+  resources :inventories do
+    resources :inventory_foods
+  end
+
+  resource :shopping_lists, only: [:index]
+  get 'recipes/:recipe_id/generate_shopping_list', :to => 'shopping_lists#new'
+  post 'recipes/:recipe_id/generate_shopping_list', :to => 'shopping_lists#create'
+  get 'shopping_list/:recipe_id/:inventory_id', :to => 'shopping_lists#index'
+  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+
+    # resources :recipes
+  # resources :recipe_foods
+  
+  # resources :inventory_foods
+  # resources :inventories
+  # resources :users
```

Comment:
- [ ] kindly remove comments to make your code base cleaner, for any queries you can refer to the previous commits.